addons/mrp/models/mrp_bom.py

```diff
@@ -180,6 +180,40 @@ def _bom_find_domain(self, product_tmpl=None, product=None, picking_type=None, c
         # order to prioritize bom with product_id over the one without
         return domain
 
+    @api.model
+    def _bom_find_multi(self, products, picking_type=None, company_id=False, bom_type=False):
+        """ Batching version of _bom_find
+
+        :param products: `product.product` recordset
+        :return: one bom (or empty recordset `mrp.bom` if none find) by product (`product.product` record)
+        :rtype: dict
+        """
+        result = {}
+        if not products:
+            return result
+        domain = [
+            '|', ('company_id', '=', False), ('company_id', '=', company_id or self.env.company.id),
+            '|', ('product_id', 'in', products.ids), '&', ('product_id', '=', False), ('product_tmpl_id', 'in', products.product_tmpl_id.ids)
+        ]
+        if picking_type:
+            domain += ['|', ('picking_type_id', '=', picking_type.id), ('picking_type_id', '=', False)]
+        if bom_type:
+            domain += [('type', '=', bom_type)]
```

Comment:
Should use `Expression.AND`, no ?
```suggestion
            domain = AND([domain, ['|', ('picking_type_id', '=', picking_type.id), ('picking_type_id', '=', False)]])
        if bom_type:
            domain = AND([domain, [('type', '=', bom_type)]])
```